Tictactoe UI

ChrisARH opened the pull request:
built a TicTacTo UI to play on browser instead of only console

varamezova commented on README.md:
````diff
@@ -5,34 +5,52 @@ This is an example implementation to query https://lebensmittelwarnung.api.bund.
+```
+
+Debug with breakpoint:
+    ```ruby
+    require 'pry';  binding.pry
+    ```
+    Find Docker image name:
+    ```Shell
+    docker ps
+    ```
+    Attach to image:
+    ```Shell
+    docker attach dockerImageName
+    ```
````
Too much indentation
~~~suggestion
```ruby
require 'pry';  binding.pry
```
Find Docker image name:
```Shell
docker ps
```
Attach to image:
```Shell
docker attach dockerImageName
```
~~~

varamezova commented on cli.rb:
```diff
@@ -4,7 +4,7 @@
 def foodwarning(amount)
     uri = URI('https://megov.bayern.de/verbraucherschutz/baystmuv-verbraucherinfo/rest/api/warnings/merged')
-    hash = {}
+    hash =  {}
```
Indentation too much
```suggestion
    hash = {}
```

varamezova commented on cli.rb:
```diff
@@ -16,7 +16,8 @@ def foodwarning(amount)
         #require 'pry'; binding.pry
         responseJSON.dig('response', 'docs').each do |warning|
-            hash[warning['title']] = [warning['warning'], warning['product']['imageUrls'].first, warning['link']]
+            #hash[warning['title']] = [warning['warning'], warning['product']['imageUrls'].first, warning['link']]
```
line can be removed
```suggestion
```

varamezova commented on forknacular.rb:
```diff
@@ -1,7 +1,71 @@
+  return "Gleichstand" if !after_arr.flatten.include?(nil)
+
+  erb(:tictactoe, locals: {field: after_arr})
+
+  #TicTacToe.valid?(before,after) ? erb(:tictactoe, locals: {field: after_arr}) : erb(:tictactoe, locals: {field: before_arr})
+ 
```
remove lines
```suggestion
```

varamezova commented on forknacular.rb:
```diff
@@ -1,7 +1,71 @@
+ 
+end
+
+
```
remove because of best practice (one line only at the end)

varamezova commented on tictactoe.rb:
```diff
@@ -95,7 +96,8 @@ def play
+    def self.valid?(before, coordinate)
+        before["before_"+coordinate] = ""
+    end
 
```
empty line between `end` & `end` can be removed (best practice)

varamezova commented on views/overview.erb:
```diff
@@ -0,0 +1,7 @@
+            <a href="/index.html">Food Warnings</a>
+            <a href="/tictactoe">Tictactoe</a>
+    </body>
+</html>
```
please add an empty line (github is complaining 🤣)

varamezova commented on views/tictactoe.erb:
```diff
@@ -0,0 +1,36 @@
+      <table border = '1'>
+        <thead>
+        <tr>
+            <th style='font-family:"Courier New", Courier, monospace; font-size:50px' colspan="3">Tictactoe</th>
```
too much indentation
```suggestion
          <th style='font-family:"Courier New", Courier, monospace; font-size:50px' colspan="3">Tictactoe</th>
```

varamezova commented on views/tictactoe.erb:
```diff
@@ -0,0 +1,36 @@
+        </tr>
+        </thead>
+        <tbody>
+          <!--was schreiben wir hier???? es ist immer noch nicht optimal! denkt mal drüber nach! vllt mit einem submit button-->
```
we have a submit button - the line can be removed

Chris-Bart committed:
debugging/style/git-lookup improvements in README

debugging instructions, syntax update, git visualisation

Chris-Bart committed:
added overview & tictactoe pages

Need for overview to discover/navigate to pages
tictactoe page does not do anything for now

Chris-Bart committed:
adapted to api changes

Chris-Bart committed:
replace depricated hash rocket notation

Chris-Bart committed:
added player move validation

will be used in next commits

Chris-Bart committed:
implement server side action

use validation for player turn
calculate computer turn and send result back to user

Azure7111 reviewed (approved):
Let's do this!
